### DISCUSSION TITLE
short constraints via "ConstraintsFunctions.assets_constraints"

### DISCUSSION THREAD

**Discussion by jdavidk** [dcajasn/Riskfolio-Lib#106] [Q&A]
If I'm looking to create a long-short portfolio (for example: upperlng = 1.3, uppersht = 0.3, budget = 1.0), and when I define the "Weight" constraints in "ConstraintsFunctions.assets_constraints", is it possible to set negative weight constraints so that certain assets are shorted more than a specific amount?

**Comment by dcajasn** [comment]
Hi @jdavidk,

Yes, you can do it, for example if you have three assets: MSFT, AMZN, AAPL; and you want that AAPL has a short weight lower than -0.3 and if you put the constraint uppersht=1 you will get the following constraints: -0.3<=AAPL<=-1, MSFT<=-1 and AMZN<=-1.

Best,
Dany
